modules/hl7802/cellular_hl7802_api.c

```diff
@@ -537,6 +550,18 @@ static CellularError_t buildSocketConfig( CellularSocketHandle_t socketHandle,
                            socketHandle->contextId,
                            socketHandle->remoteSocketAddress.ipAddress.ipAddress,
                            socketHandle->remoteSocketAddress.port );
+
+        /* Set the local port in the end of command buffer string if localPort is not 0. */
+        if( socketHandle->localPort > 0 )
+        {
+            ( void ) snprintf( portBuf, sizeof( portBuf ),
+                               ",%u",
+                               socketHandle->localPort );
+
+            /* Because the length of host's IP address is limited,
+             * the buffer size must be enough for port setting */
```

Comment:
Would an assert be useful here?

Reply:
Reserve buffer at the previous step to guarantee that.